### PR TITLE
[Composable API] `replicate`: cleanup _ddp.py

### PR DESCRIPTION
**Pull request by yhcharles**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #90257

### CODE REVIEW

File: torch/distributed/_composable/_ddp.py [hunk 1]
```diff
@@ -179,368 +92,15 @@ def backward(ctx, *grad_outputs):
...
-        module (Module): the module to be parallelized.
-
-    Example::
-
-        >>> # xdoctest: +SKIP("undefined variables")
-        >>> torch.distributed.init_process_group(backend='nccl', world_size=4, init_method='...')
-        >>> net = torch.nn.parallel.DistributedDataParallel(model)
-    """
-
+class DistributedDataParallel(Module):
```
Comment (mrshenli) [inline torch/distributed/_composable/_ddp.py:103]: I saw the `replicate` API calls into this module wrapper. Is this necessary? Do you plan to extract states and method from this file and remove the model wrapper later?

```
        self._ddp = _ddp.DistributedDataParallel(
            self._param_list, **self.kwargs
        )
```

**Review by mrshenli — APPROVED** [review]

File: torch/distributed/_composable/_ddp.py (hunk 1 above)
Comment (yhcharles) [reply torch/distributed/_composable/_ddp.py:103]: Yes, will do!

**Pull request closed without merging**